### PR TITLE
PE 1111/Handle gql errors

### PR DESCRIPTION
**Pull request by javdhu**
Handles errors after retry fails at login and emits failure states.

### CODE REVIEW

File: lib/blocs/profile_add/profile_add_cubit.dart [hunk 1]
```diff
@@ -56,8 +57,15 @@ class ProfileAddCubit extends Cubit<ProfileAddState> {
...
     _wallet = Wallet.fromJwk(json.decode(walletJson));
-    _driveTxs =
-        await _arweave.getUniqueUserDriveEntityTxs(await _wallet.getAddress());
+
+    List<TransactionCommonMixin> _driveTxs;
+    try {
+      _driveTxs = await _arweave
+          .getUniqueUserDriveEntityTxs(await _wallet.getAddress());
+    } catch (e) {
+      emit(ProfileAddFailiure());
```
Comment (thiagocarvalhodev) [inline lib/blocs/profile_add/profile_add_cubit.dart:66]:
```suggestion
      emit(ProfileAddFailure());
```

**Review by thiagocarvalhodev — APPROVED** [review]
LGTM. Left only a typo fix.

**Review by karlprieb — COMMENTED** [review]
Can we update `ProfileAddCubit` tests to catch `ProfileAddFailure`?

**Review by karlprieb — DISMISSED** [review]

**Review by thiagocarvalhodev — APPROVED** [review]

**Review by karlprieb — APPROVED** [review]

File: lib/blocs/profile_add/profile_add_cubit.dart [hunk 2]
```diff
@@ -132,7 +149,7 @@ class ProfileAddCubit extends Cubit<ProfileAddState> {
               !(await arconnect.checkPermissions()))) {
```
Comment (matibat) [inline lib/blocs/profile_add/profile_add_cubit.dart:149]: So it's in the above line where _lastKnownWallerAddress might be null

File: lib/utils/graphql_retry.dart [hunk 3]
```diff
@@ -11,15 +12,21 @@ class GraphQLRetry {
   final ArtemisClient _client;
 
   Future<GraphQLResponse<T>> execute<T, U extends JsonSerializable>(
-      GraphQLQuery<T, U> query,
-      {Function(Exception e)? onRetry}) async {
+    GraphQLQuery<T, U> query, {
+    Function(Exception e)? onRetry,
+    int maxAttempts = 8,
```
Comment (matibat) [inline lib/utils/graphql_retry.dart:17]: We did have a constant value for it we can re-use here

**Review by matibat — APPROVED** [review]
Left a comment and a non-blocking nit suggestion

**Pull request merged**